diff -ur tse3-0.3.1.orig/src/tse3/plt/Alsa-0.9.cpp tse3-0.3.1/src/tse3/plt/Alsa-0.9.cpp
--- tse3-0.3.1.orig/src/tse3/plt/Alsa-0.9.cpp	2005-07-25 13:22:56.000000000 +0200
+++ tse3-0.3.1/src/tse3/plt/Alsa-0.9.cpp	2007-11-30 20:51:03.773570339 +0100
@@ -30,7 +30,7 @@
 #include <errno.h>
 #include <alloca.h>
 #if HAVE_ALSA_ASOUNDLIB_H
-#include <sys/asoundlib.h>
+#include <alsa/asoundlib.h>
 #elif HAVE_SYS_ASOUNDLIB_H
 #include <sys/asoundlib.h>
 #endif
